app/controllers/cloud_object_store_container_controller.rb

```diff
@@ -17,7 +17,12 @@ def button
     @edit = session[:edit] # Restore @edit for adv search box
     params[:page] = @current_page unless @current_page.nil? # Save current page for list refresh
 
-    process_cloud_object_storage_buttons(params[:pressed])
+    case params[:pressed]
+    when "cloud_object_store_container_new"
+      return javascript_redirect(:action => "new")
+    else
```

Comment:
Thanks for moving the `new` button handling in this controller.
Next, we would be adding the `edit` case here.